sdk/helper/keysutil/policy.go

```diff
@@ -2393,3 +2397,132 @@ func wrapTargetPKCS8ForImport(wrappingKey *rsa.PublicKey, preppedTargetKey []byt
 	wrappedKeys := append(ephKeyWrapped, targetKeyWrapped...)
 	return base64.StdEncoding.EncodeToString(wrappedKeys), nil
 }
+
+func (p *Policy) CreateCsr(keyVersion int, csrTemplate *x509.CertificateRequest) ([]byte, error) {
+	keyEntry, err := p.safeGetKeyEntry(keyVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	if keyEntry.IsPrivateKeyMissing() {
+		// NOTE: Shouldn't this be a client error?
+		return nil, errutil.UserError{Err: "private key not imported for key version selected"}
+	}
+
+	csrTemplate.Signature = nil
+	csrTemplate.SignatureAlgorithm = x509.UnknownSignatureAlgorithm
+
```

Comment:
Wouldn't it be better to leave these overrides here just in case the template is signed? I realized this when I tried removing these lines and rerun the tests, as the template CSR used as base was signed with an RSA key and raised errors on the other key types tests.

Reply:
Hmm, this is OK then. For certificates, I thought Go would override an unsupportable recommendation with the default of the key algorithm, but it looks like I was mistaken and we (Vault) actually do that work in certutil. My bad!